cat-launcher/cat-macros/src/lib.rs

```diff
@@ -0,0 +1,39 @@
+extern crate proc_macro;
+
+use proc_macro::TokenStream;
+use quote::quote;
+use syn::{parse_macro_input, DeriveInput};
+
+#[proc_macro_derive(CommandErrorSerialize)]
+pub fn derive_command_error_serialize(input: TokenStream) -> TokenStream {
+    // Parse the input tokens into a syntax tree
+    let input = parse_macro_input!(input as DeriveInput);
+
+    // Get the name of the type
+    let name = &input.ident;
+    let name_str = name.to_string();
+
+    // Generate the implementation of the Serialize trait
+    let expanded = quote! {
+        impl ::serde::Serialize for #name {
+            fn serialize<S>(&self, serializer: S) -> ::std::result::Result<S::Ok, S::Error>
+            where
+                S: ::serde::Serializer,
+            {
+                use ::serde::ser::SerializeStruct;
+                let mut st = serializer.serialize_struct(#name_str, 2)?;
+
+                let err_type: &'static str = self.into();
```

Comment:
In cat-launcher/cat-macros/src/lib.rs around lines 26 to 27, the generated code
calls self.into() assuming &Self implements Into<&'static str> but the macro
doesn't enforce that; add a where-clause to the generated impl (for example
require Self: strum::IntoStaticStr or equivalently for<'a> &'a Self:
Into<&'static str>) so the compiler emits a clear "trait bound not satisfied"
error when the user forgets to derive IntoStaticStr, leaving the self.into()
usage unchanged.